controller/linodemachine_controller.go

```diff
@@ -287,6 +301,33 @@ func (r *LinodeMachineReconciler) reconcileCreate(
 	return ctrl.Result{}, nil
 }
 
+func (r *LinodeMachineReconciler) reconcilePreflightLinodeFirewallCheck(ctx context.Context, logger logr.Logger, machineScope *scope.MachineScope) (ctrl.Result, error) {
```

Comment:
Thats a good point. We would like to keep our linode API calls as minimum as possible. Since this falls in linodemachine_controller, doing GET before provisioning would mean we would be doing this for each instance, reconciles on error would mean we'll be redoing the calls and I feel we can simply avoid it and rely on k8s having the linodefirewall ready/not-ready to decide if firewall is available or not. We do search based on firewall name, a typo from customer would mean firewall not found and will get stuck with firewall not ready condition. In worst case if it matches with another linode firewall in cluster (same name and in same namespace which has a very low probability), thats a customer configuration and I would expect us to honor it. Hence leaning towards not making an API call here.

Reply:
Yeah agreed. Sounds good to leave as is 👍